pywbem_mock/_instancewriteprovider.py

```diff
@@ -270,192 +213,231 @@ def __init__(self, cimrepository=None):
-                     if 'key' in p.qualifiers]
-        for pn in key_props:
-            if pn not in new_instance:
-                raise CIMError(
-                    CIM_ERR_INVALID_PARAMETER,
-                    _format("Key property {0!A} not in NewInstance ", pn))
-
-        # Exception if property in instance but not class or types do not
-        # match
-        validate_inst_props(namespace, target_class, new_instance)
-
-        # Build instance path from values in the instance.
-        new_instance.path = CIMInstanceName.from_instance(
-            target_class,
-            new_instance,
-            namespace=namespace)
-
-        # Store the new instance in the  CIM repository if it is not
-        # already in the repository
+        # Store the new instance in the CIM repository, verifying that it does
+        # not exist yet.
         try:
-            instance_store.create(new_instance.path, new_instance)
+            instance_store.create(NewInstance.path, NewInstance)
         except ValueError:
             raise CIMError(
                 CIM_ERR_ALREADY_EXISTS,
-                _format("NewInstance {0!A} already exists in namespace "
-                        "{1!A}.", new_instance.path, namespace))
+                _format("New instance {0!A} already exists in namespace "
+                        "{1!A}.", NewInstance.path, namespace))
 
-        # Create instance returns model path
-        return new_instance.path
+        # CreateInstance returns the instance path of the new instance
+        return NewInstance.path
 
     ####################################################################
     #
-    #   ModifyInstance server request
+    #   ModifyInstance mock WBEM server responder
     #
     ####################################################################
 
     def ModifyInstance(self, ModifiedInstance,
                        IncludeQualifiers=None, PropertyList=None):
         # pylint: disable=invalid-name,line-too-long,unused-argument
         """
-        Implements a WBEM server responder for
+        Implements a mock WBEM server responder for
         :meth:`pywbem.WBEMConnection.CreateInstance`.
 
-        Modify a CIM instance in the CIM repository.
-
-        NOTE: This method includes namespace within the path element
-        of the ModifiedInstance rather than as a separate input parameter.
+        Modify an existing CIM instance in the CIM repository of the mock WBEM
+        server.
+
+        NOTE: This method specifies the namespace in ModifiedInstance.path
+        rather than as a separate input parameter.
+
+        The modification of the instance in the CIM repository that is performed
+        by the provider should be limited to property value changes (including
+        addition of properties if not yet set on the instance), because
+        instance-level qualifiers have been deprecated in CIM.
+
+        The set of properties that are to be modified on the CIM instance, is:
```

Comment:
Think in the CIM Instance better than on the CIM instance

Reply:
DONE